cluster/query_executor.go

```diff
@@ -320,6 +320,16 @@ func (e *QueryExecutor) executeDropSeriesStatement(stmt *influxql.DropSeriesStat
 	return e.TSDBStore.DeleteSeries(database, stmt.Sources, stmt.Condition)
 }
 
+func (e *QueryExecutor) executeDropShardStatement(stmt *influxql.DropShardStatement) error {
+	// Remove the shard reference from the Meta Store.
+	if err := e.MetaClient.DropShard(stmt.ID); err != nil {
```

Comment:
@jwilder :
> Should this come after the TSDBStore delete call? If the TSDBStore delete fails for some reason, would you be able to retry and clean it up again?

(sorry I think I deleted your comment by mistake). The meta store won't return an error if the shard does not exist, so it should be safe to call both these methods repeatedly.